elixir/lib/contracts.ex

```diff
@@ -69,7 +90,30 @@ defmodule Trento.Contracts do
     end
   end
 
-  defp decode(type, data) do
+  defp decode_and_validate(value, validate_expiration) do
+    with {:ok, event_type, attributes, event_data} <- extract_event(value),
+         {:ok, decoded_event} <- decode_trento_event(event_type, event_data),
+         :ok <- maybe_validate_expiration(attributes, !validate_expiration) do
```

Comment:
Do you think we can remove the negation here? 
Having `true` in the function usage if we need to validate is more intuitive.

You just need to replace with this: `defp maybe_validate_expiration(_payload, false), do: :ok`
nitpick: you could even put `true` instead of `_skip_validation`, it looks more representative

Reply:
done!